.github/CONTRIBUTING.md

```diff
@@ -1,36 +1,169 @@
+# Contributing
+### Anyone is welcome to make BeEF better!
+Thank you for wanting to contribute to BeEF. It's effort like yours that helps make BeEF such a great tool.
 
-4. Make a pull request from the forked branch to BeEF master
+Following these guidelines shows that you respect the time of the developers developing this open source project and helps them help you. In response to this, they should return that respect in addressing your issue, assisting with changes, and helping you finalize your pull requests.
 
-See https://github.com/beefproject/beef/wiki/BeEF-Testing for steps to write tests.
 
-TL;DR: 
-```
-bundle install --with test
-bundle exec rake spec
-```
+###  We want any form helpful of contributions!
 
-4. Write tests in RSpec for your new code (module, extension etc.)
+ BeEF is an open source project and we love to receive contributions from the community! There are many ways to contribute, from writing tutorials or blog posts, improving or translating the documentation, answering questions on the project, submitting bug reports and feature requests or writing or reviewing code which can be merged into BeEF itself.
 
-5. Run all tests again to make sure they all pass
 
-6. Edit existing wiki page / add a new one explaining the new features, including:
+
+# Ground Rules
+
+###  Responsibilities
+> * When making an issue, ensure the issue template is filled out, failure to do so can and will result in a closed ticket and a delay in support.
+> * We now have a two-week of unresponsiveness period before closing a ticket, if this happens, just comment responding and @ whoever closed the ticket. 
+> * Ensure cross-platform compatibility for every change that's accepted. Mac and Linux are currently supported.
+> * Create issues for any major changes and enhancements that you wish to make. Discuss things transparently and get community feedback.
+> * Ensure language is as respectful and appropriate as possible. 
+> * Keep merges as straightforward as possible, only address one issue per commit where possible.
+> * Be welcoming to newcomers and try to assist where possible, everyone needs help. 
+
+# Where to start
+### Looking to make your first contribution 
+
+ Unsure where to begin contributing to BeEF? You can start by looking through these issues:
+
+ * Good First Issue - issues which should only require a few changes, and are good to start with.
+ * Question - issues which are a question and need a response. A good way to learn more about BeEF is to try to solve a problem.
+
+At this point, you're ready to make your changes! Feel free to ask for help; everyone is a beginner at first.
+
+If a maintainer asks you to "rebase" your PR, they're saying that code has changed, and that you need to update your branch so it's easier to merge.
+
+### Ruby best practise 
+Do read through: https://rubystyle.guide
+Try and follow through with the practices throughout, even going through it once will help keep the codebase consistent.
```

Comment:
It would be a good idea to mention Rubocop somewhere.

Ideally, BeEF should ship a `.rubocop` file which specifies the expected style.

Of course, in practicing what you preach, the entire codebase should first be updated for consistency.

I don't care about JS style and I'm not sure if there's any linting tools that we can feasible use (due to having to support old browsers).

Reply:
Thats a good idea, i will add it below the ruby style guide.
We actually already do ship a rubocop file that i believe jackdwalker configured. I am unsure about the configuration itself but i know its on the laxer side of things and picks up a good 20100 things already. I think we what beef needs to use there may be some sections where we cant help it but that needs to be actioned, I'll raise a ticket for it and try to jump on at least some of it, ill go to rubo -v 0.57 and see how many of them can be automatically fixed aka formatting errors.
We may wanna introuce a version in the gemfile tho as its missing and pick a version to stick to (as they have breakable changes).
I think ESLint is an alright javascript one but i also don't know about how that would work either, i think there is value in investigating(i'll raise an issue) but i also believe rubocop should be focused on first.

Reply:
> We actually already do ship a rubocop file

Oh right. I forgot I added it. Yes, there are many violations.